_books/ion-1-1/src/glossary.md

```diff
@@ -0,0 +1,160 @@
+like _module-name_::_name-or-address_.
+
+**required parameter**<br/>
+A macro parameter that is not _optional_ and therefore requires an argument at each invocation.
+
+**rest parameter**<br/>
+A macro parameter—always the final parameter—declared with `*` or `+` cardinality,
+that accepts all remaining individual arguments to the macro as if they were in an implicit _argument group_.
+Applies to Ion text and TDL.
+Similar to "varargs" parameters in Java and other languages.
+
+**segment**<br/>
+A contiguous partition of a _document_ that uses the same _active encoding module_.
+Segment boundaries are caused by directives: an IVM starts a new segment (ending the prior segment, if any),
+while `$ion_symbol_table` and `$ion_encoding` directives end segments (with a new one starting immediately afterward).
+
+**shared module**<br/>
+A module that exists independent of the data stream of an Ion document. It is identified by a
+name and version so that it can be imported by other modules.
+
+**signature**<br/>
+The part of a macro definition that specifies its "calling convention", in terms of the shape,
+type, and cardinality of arguments it accepts.
+
+**symbol table directive**<br/>
+A top-level struct annotated with `$ion_symbol_table`.  Defines a new encoding environment
+without any macros.  Valid in Ion 1.0 and 1.1.
+
+**system e-expression**<br/>
+An _e-expression_ that invokes a _macro_ from the _system-module_ rather than from the _active encoding module_.
+
+**system macro**<br/>
+A macro provided by the Ion implementation via the system module `$ion`.
+System macros are available at all points within Ion 1.1 segments.
+
+**system module**<br/>
+A standard module named `$ion` that is provided by the Ion implementation, implicitly installed so
+that the system symbols and system macros are available at all points within a document.
+Subsumes the functionality of the Ion 1.0 system symbol table.
+
+**system symbol**<br/>
+A symbol provided by the Ion implementation via the system module `$ion`.
+System symbols are available at all points within an Ion document, though the selection of symbols
+varies by segment according to its Ion version.
+
+**TDL**<br/>
+See _template definition language_.
+
+**template**<br/>
+The part of a macro definition that expresses its transformation of inputs to results.
+
+**template definition language**<br/>
+An Ion-based, domain-specific language that declaratively specifies the output produced by a _macro_.
+Template definition language uses only the Ion data model (i.e. valid Ion 1.0 syntax).
+
+**unqualified macro reference**<br/>
+A macro reference that consists of either a macro name or numeric address, without a qualifying module name. 
+These are resolved using lexical scope and must always be unambiguous.
+
+**variable expansion**<br/>
+In _TDL_, a special form that causes the expanded _arguments_ for the given _parameter_ to be substituted into the _template_.
```

Comment:
There are some mixed plural/singular usages of "argument" that make its meaning a bit hazy. Based on usage, a reader could understand it to mean either:
1. a single expression (possibly one of a group) passed to a parameter
  OR
2. the only expression passed to a parameter (this expression may be an expression group containing subexpressions)

We should consolidate our usage to reflect the intended meaning.

Reply:
```suggestion
In _TDL_, a special form that causes all argument expression(s) for the given _parameter_ to be expanded and the result of the expansion to be substituted into the _template_.
```